Automattic/wp-calypso
# Submit Tracks events on filter updates in sites dashboard

Fixes https://github.com/Automattic/wp-calypso/issues/94433

## Proposed Changes

We currently don't submit any Tracks event for the `Status` filter on wordpress.com/sites. This PR fixes that.

There's currently only one filterable field in the sites dashboard, but I've taken extra steps to use TS types that prompt us to consider this aspect when if/when new fields are added.

## Why are these changes being made?


This data would be useful when considering new projects, like for example https://github.com/Automattic/dotcom-forge/issues/8010

## Testing Instructions



1. Navigate to `/sites`
2. Select a value for the `Status` filter
3. Ensure that a `calypso_sites_dashboard_filter_status` Tracks event is logged with the correct `value` prop
4. Enter something in the search field
5. Ensure that no new `calypso_sites_dashboard_filter_status` Tracks events were logged
6. Change the value of the `Status` filter
7. Ensure that a new `calypso_sites_dashboard_filter_status` Tracks event is logged with the correct `value` prop

## Pre-merge Checklist
